src/Mod/Part/Gui/Command.cpp

```diff
@@ -374,12 +374,15 @@ void CmdPartCommon::activated(int iMsg)
         numShapes = 1; //to be updated later in code, if
         Gui::SelectionObject selobj = Sel[0];
         TopoDS_Shape sh = Part::Feature::getShape(selobj.getObject());
-        if (sh.ShapeType() == TopAbs_COMPOUND) {
+        while (numShapes==1 and sh.ShapeType() == TopAbs_COMPOUND) {

```

Comment:
Don't use the keyword `and`! By default, this doesn't work with every compiler.

Reply:
ok fusion actually does fail with a single shape only.  common doesn't care (since it treats the compound of multiple shapes as a single shape anyway) there only is a warning

so I'm gonna remove the duplicated code from common and thus the code duplication

Reply:
> I did not make a function for this, since I simply had no clue which class and folder to put it into.

E.g. a function can be added to TopoShape: `static int TopoShape::countShapes(const TopoDS_Compound&)`

> Also this is GUI code, I don't know how to put a test case for this.

It's used in the GUI but it's no GUI code.

> I now added tests for Part::MultiFuse with compounds and recursive compounds.

Actually, there is too much high-level code involved. Ideally, you only use the OCC API to construct a nested compound and pass this shape to `TopoShape::countShapes`.

Reply:
yeah that could be done.

since it's currently only used in this one place, would you mind separating that from this issue/PR and making a separate issue/TODO for it? that way this bugfix doesn't have to wait until I get to implement that, kinda limited on time atm...

Reply:
I opened an issue for this so it doesnt get forgotten #17506 17506